File: web/apps/web-docs-hub/docs/examples/hello_world.go
// Igris Inertial — Go hello world
// Requires: go get github.com/igris-inertial/go-sdk
//
// Usage:
//   export IGRIS_API_KEY=igris_...
//   go run hello_world.go

package main

import (
	"context"
	"fmt"
	"log"
	"os"

	igris "github.com/igris-inertial/go-sdk"
)

func main() {
	apiKey := os.Getenv("IGRIS_API_KEY")
	if apiKey == "" {
		log.Fatal("Set IGRIS_API_KEY first: export IGRIS_API_KEY=igris_...")
	}

	client := igris.NewClient("https://overture.igrisinertial.com", apiKey)
	ctx := context.Background()

	resp, err := client.Infer(ctx, &igris.InferRequest{
		Model: "gpt-4",
		Messages: []igris.Message{
			{Role: "user", Content: "Hello! Reply in one sentence."},
		},
	})
	if err != nil {
		log.Fatalf("infer: %v", err)
	}

	fmt.Println(resp.Choices[0].Message.Content)
	if resp.Metadata != nil {
		fmt.Printf("Provider : %s\n", resp.Metadata.Provider)
		fmt.Printf("Latency  : %d ms\n", resp.Metadata.LatencyMs)
		fmt.Printf("Cost     : $%.6f\n", resp.Metadata.CostUSD)
	}
}

File: web/apps/web-docs-hub/docs/examples/hello_world.rs
// Igris Inertial — Rust hello world
// Requires: cargo add igris-inertial
//
// Usage:
//   export IGRIS_API_KEY=igris_...
//   cargo run

use igris_inertial::{IgrisClient, InferRequest, Message};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let api_key = std::env::var("IGRIS_API_KEY")
        .expect("Set IGRIS_API_KEY first: export IGRIS_API_KEY=igris_...");

    let client = IgrisClient::builder("https://overture.igrisinertial.com")
        .api_key(api_key)
        .build()?;

    let response = client
        .infer(&InferRequest {
            model: "gpt-4".to_string(),
            messages: vec![Message {
                role: "user".to_string(),
                content: "Hello! Reply in one sentence.".to_string(),
                ..Default::default()
            }],
            max_tokens: Some(64),
            ..Default::default()
        })
        .await?;

    println!("{}", response.choices[0].message.content);
    if let Some(meta) = &response.metadata {
        println!("Provider : {}", meta.provider);
        println!("Latency  : {} ms", meta.latency_ms);
        println!("Cost     : ${:.6}", meta.cost_usd);
    }

    Ok(())
}
